File: 프로그래머스/lv3/49191. 순위/순위.py
def solution(n, results):
    answer = 0
    board = [[0] * n for _ in range(n)]
    
    for a, b in results:
        board[a-1][b-1] = 1
        board[b-1][a-1] = -1

    for i in range(n):
        for j in range(n):
            for k in range(n):
                if board[j][i] == 0:
                    continue

                if board[j][i] == board[i][k]:
                    board[j][k] = board[j][i]
                    board[k][j] = -board[j][k]

    for i in range(n):
        if board[i].count(0) == 1:
            answer += 1

    return answer

File: 프로그래머스/3/49191. 순위/순위.java
class Solution {
    public int solution(int n, int[][] results) {
        int answer = 0;

        int[][] arr = new int[n + 1][n + 1];

        for (int[] result : results) {
            arr[result[0]][result[1]] = 1;
            arr[result[1]][result[0]] = -1;
        }

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                for (int k = 1; k <= n; k++) {
                    if (arr[i][k] == 1 && arr[k][j] == 1) {
                        arr[i][j] = 1;
                        arr[j][i] = -1;
                    }
                    if (arr[i][k] == -1 && arr[k][j] == -1) {
                        arr[i][j] = -1;
                        arr[j][i] = 1;
                    }
                }
            }
        }

        for (int i = 1; i <= n; i++) {
            int cnt = 0;
            for (int j = 1; j <= n; j++) {
                if (arr[i][j] != 0) {
                    cnt++;
                }
            }
            if (cnt == n - 1) {
                answer++;
            }
        }

        return answer;
    }
}